include/linux/seqlock.h

```diff
@@ -555,22 +555,27 @@ static inline void read_seqbegin_or_lock(seqlock_t *lock, int *seq)
 
 static inline int need_seqretry(seqlock_t *lock, int seq)
 {
-	int ret;
+	return !(seq & 1) && read_seqretry(lock, seq);
```

Comment:
This is how the function looks in vanilla kernel. I am actually reverting my previous changes to this function.
As for the need_seqretry_check, ktsan_seqcount_begin is no-op in normal build, so compiler should be able to generate decent code without tsan.